crates/compute/src/cpu/layer.rs

```diff
@@ -7,14 +7,72 @@ use binius_field::{
 };
 
 use super::{memory::CpuMemory, tower_macro::each_tower_subfield};
-use crate::layer::{ComputeLayer, Error, FSlice, FSliceMut};
+use crate::{
+	layer::{ComputeLayer, Error, FSlice, FSliceMut},
+	memory::FSliceWithTowerLevel,
+};
 
 #[derive(Debug)]
 pub struct CpuExecutor;
 
 #[derive(Debug, Default)]
 pub struct CpuLayer<F: TowerFamily>(PhantomData<F>);
 
+/// evals is treated as a matrix with `1 << log_query_size` rows and each column is dot-producted
```

Comment:
See https://github.com/IrreducibleOSS/binius/blob/main/CONTRIBUTING.md#documentation for docstring formatting.